### DISCUSSION TITLE
How can you tell if WZ_mini has booted?

### DISCUSSION THREAD

**Discussion by mmarlette** [gtxaspec/wz_mini_hacks#454] [Q&A]
This is a newbie question but I have tried everything I can think of and read most of the threads but nothing seems to work.

I have multiple new V3 cameras, one is at 4.36.9.139 and the other is right out of the box, no update is at 4.36.3.19

I have added both cameras via the Wyze app and didn't update them when the app prompted me.

Went to the installed camera in the Wyze app, settings, device info, to get the IP address of the camera, which is 192.168.254.33

Both cams view under Wyze app.

My root of the 32GB SD card formatted as FAT32 has the following in the root of the drive:
System Volume Information
wz_mini
factory_t31_ZMC6tiIDQN

I have added to the wz_mini.conf file the following:
RTSP_LOGIN="admin"
RTSP_PASSWORD="1234"
RTSP_PORT="8554"

RTSP_HI_RES_ENABLED="true"
RTSP_HI_RES_ENABLE_AUDIO="true"
RTSP_HI_RES_MAX_BITRATE="2048"
RTSP_HI_RES_TARGET_BITRATE="1024"
RTSP_HI_RES_ENC_PARAMETER="2"
RTSP_HI_RES_FPS=""

RTSP_LOW_RES_ENABLED="false"
RTSP_LOW_RES_ENABLE_AUDIO="false"
RTSP_LOW_RES_MAX_BITRATE=""
RTSP_LOW_RES_TARGET_BITRATE=""
RTSP_LOW_RES_ENC_PARAMETER=""
RTSP_LOW_RES_FPS=""

RTSP_AUTH_DISABLE="false"
-----
On win10, start VLC, open network stream,  enter url as :rtsp://admin:1234@192.168.254.33:8554/unicast
Connection Failed

I have also tried RTSP_AUTH_DISABLE="true", same response.

I have also installed the SSH public key in the SD card path, wz_mini/etc/ssh/authorized_keys. I have my private key in the c:\users\Mark\.ssh directory, with the private key added to the SSH service.

When I use the command:
ssh -T 192.168.254.33
ssh: connect to host 192.168.254.33 port 22: Connection refused

When I try to putty into Host IP of 192.168.254.33 port 22, connection type SSH. I get Network Error: Connection refused.

I do have V2's running under Defang, so I have some ability... :) I am starting to wonder.

The goal is to get the wireless connection to the V3 working, then to get the POE wired Ethernet connection working under Blue Iris, with the rest of my cameras.

I have to be missing something ...I am not looking to be spoon fed but a good slap or push in the right direction.

TIA, Mark

**Comment by Bob-AOL** [comment]
Additional files will be created on the SD card upon a successful boot/load. System Volume Information should not be included on the SD card. If indeed it is, reformat with the wz_mini folder and the factory camera I.D. as the ONLY contents. 

You have no entry for FPS in the RTSP section of wz_mini.conf. Not sure it matters for wz_mini to simply load. I'm guessing it may for the RTSP server to establish a stream. 20 is the default for the v3s.

I struggled with SSH in Windows until simply activating the Open SSH Client for Windows. Very simple and reliable solution once you generate a key and become familiar with it's command line interaction with the camera. Also easy to troubleshoot if it misbehaves.

When it comes to activating the USB/ETH functionality, I can certainly provide additional clarification to what is provided in the README.
Plug and Play auto-detect as is implied it is not.

**Reply by mmarlette** [reply]
Thank you for the reply, very much appreciate it.
I did what you indicated above, which I have done several times and it now booted, as there is a different voice upon boot. It obviously wanted something.
I then entered VLC, open network  stream, rtsp://admin:1234@192.168.254.33:8554/unicast
Presto..... it appeared.
I must not have my SSH keys correct as I am not able to SSH to the camera yet. Looking at that now.

**Comment by Bob-AOL** [comment]
Just remembered VLC prefers the Wyze format and will prompt for login credentials -- rtsp://192.168.254.33:8554/live is what it wants. It will then ask for login information. This convention works for me, but I'm also running 4.61.0.3 Wyze RTSP BETA firmware which specifies that format in the actual Wyze app itself. Not sure that affects anything, but Blue Iris will also use it by default, albeit as an http v. rtsp URL.

You may want to flash the Wyze RTSP firmware and go from there. You will at least be able to fall back to it for live streaming in the event of further difficulties with wz_mini.

I also use RTSP port 554..., just because.

**Reply by mmarlette** [reply]
I have the Wyze RTSP on another camera, that is not powered ATM. 

It works under Blue Iris and the Wyze app no problem.

The Wyze RTSP doesn't support wired USB/ETH functionality, does it? I haven't gotten that far. I do have the hardware for a Realtek 8152 setup.

Thaks again, GREAT information.

**Reply by Bob-AOL** [reply]
You'll need wz_mini to implement the USB/ETH functionality. Are you intending to use the UCTRONICS adapter? (It's a Realtek 8152 device.) They're great when up and running and have been bullet proof for me despite the numerous complaints of 'bad ones'. Any difficulties I've had with them have not been the fault of the adapters, but rather from not understanding how to properly set them up successfully.

I'm almost finished with a tutorial about the considerations involved/required for implementing the USB/ETH functionality with particular respect to the UCTRONICS adapter for Windows users. It's definitely a tl;dr for most folks, but covers quite a bit of territory for the less experienced who find themselves here wanting to learn about the wz_mini_hacks buzz and maybe take it for a spin around the block.

Be more than happy to share the more relevant points if you need assistance or simply want to save some headbanging.

**Reply by mmarlette** [reply]
Ok, I have SSH working under putty and powershell ssh with no paraphrase typing each time.

Yes, UCTRONICS for Pi. It is a white box with RJ45 jack/plug and a USB min for the POE power input.

I opened it up as I couldn't find the chipset listed anywhere and verified it was the supported Realtek 8152 chipset for WZ_mini.

I will be starting this tomorrow. I am all ears on anything you want to share. :)

Appreciate your assistance so far. Some of the other threads helped on the SSH, which I have never done before with keys. Nice.

**Reply by Bob-AOL** [reply]
So I've almost completed my 'thesis' but don't want to step on any toes posting it in its entirety until running it past the developer. If you'd care to provide contact info I'd be more than happy to shoot it your way for both a cursory review and hopefully some assistance in getting your USB/POE/ETH up and running smoothly.

**Reply by mmarlette** [reply]
Appreciate the consideration.

mark@cloud9tech.com

**Reply by Bob-AOL** [reply]
I enjoy helping when and if I can. Incoming shortly. Grab a cup of coffee and a sandwich. LOL

**Reply by Bob-AOL** [reply]
Just a correction to my reference above regarding the MRL that VLC viewer wants to see when using the RTSP Beta 4.61.0.3 firmware. It will _only_ work with port 554 and the /live MRL address from the Wyze RTSP server when using the Beta firmware. The wz_mini server appears to be overridden by the hard coded Wyze server as discovered and generally documented in the recent discussion over in #458 . This only applies to the RTSP Beta firmware. Stock Wyze firmware uses the wz_mini RTSP server and the /unicast MRL(s) and the port can be changed if desired.

Rather surprising that this subject hasn't received more visibility -- at least for the general masses.
